slitlessutils/core/preprocess/astrometry/affinetweak.py

```diff
@@ -1,85 +1,160 @@
 from astropy.io import fits
 import numpy as np
-import os
+
+from ....logger import LOGGER
+from . import utils
 
 
 class AffineTweak(dict):
+    """
+    Class to contain an affine perturbation to the Astrometry of a fits image
+    between two different WCS solutions present in a file.
+
+    This assumes the "classic" astrometric formulation of a CD (or PC) matrix
+    and a CRVAL vector.  The affine transformation is a simple vector difference
+    of the CRVAL keywords and a matrix product between the CD matrices.  This
+    permits for skew, rotation, and translation between the two WCSs.  Now, the
+    method computes the multiplicative matrix of the CD-matrix and vector
+    difference between the final WCS and the initial WCS:
+
+    .. math::
+       A = CD_{0} CD_{1}^{-1}
+
+    This is then matrix-multiplied into the CD matrix for a different image to
+    update it.  In general, this matrix A is not simply a rotation matrix as
+    the astrometric tweaks may include skew or pixel-size changes.  In a similar
+    fashion, the difference in the CRVALs between the final and initial WCSs:
+
+    .. math::
+      \\Delta CRVAL = CRVAL_{1} - CRVAL_{0}
+
+    In general, these tweaks are typically <~0.1 arcsec.
+
+
+    Parameters
+    ----------
+    reffile : str
+        The file name of the image to use as a reference.
+
+    key0 : str, optional
+        The WCS key for the initial WCS.  This is usually the "older" WCS.
+        Note, this is not case-sensitive (only capitals are used).
+        Default is "A"
+
+    key1 : str, optional
+        The WCS key for the final WCS.  This is usually the "newer" WCS.
+        Note, this is not case-sensitive (only capitals are used).
+        Default is '' (the empty string).
+
+    Notes
+    -----
+    This class inherits from ``dict`` to store individual tweaks for each
+    detector, in the case of instruments that have multiple detectors (e.g.
+    ACS WFC).  The keyword/value pairs for the dict will be the extension
+    name/version and a dict of data for the perturbations.  Users are
+    highly discouraged from manually manipulating those contents.
+
+    Examples
+    --------
+    The usual work flow would be to pass a direct image that has been
+    tweaked to some optimal solution (e.g. referenced to Gaia) to derive
+    the perturbations.  Then apply those perturbations to a grism image
+    that has not been similarly updated.
+
+    >>> tweak = AffineTweak('directimage_flt.fits')
+    >>> newfile = tweak('grismimage_flt.fits')
```

Comment:
```suggestion
    >>> tweak = AffineTweak('directimage_flt.fits') # doctest: +SKIP
    >>> newfile = tweak('grismimage_flt.fits') # doctest: +SKIP
```

Need this so the tests will pass.